### DISCUSSION TITLE
Azure Active Directory Group Claim

### DISCUSSION THREAD

**Discussion by chris468** [OpenIDC/mod_auth_openidc#599] [Q&A] [answered]
With Azure Active Directory `groups` will be returned as a [distributed claim](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#groups-overage-claim) when it is too long. When that happens, authorization based on the `groups` claim doesn't work. I believe mod_auth_openidc is checking for distributed claims returned from the UserInfo endpoint, but not in the id_token. And it [doesn't look like AAD can be configured to return the groups claim from UserInfo](https://docs.microsoft.com/en-us/azure/active-directory/develop/userinfo#notes-and-caveats-on-the-userinfo-endpoint).

Is there a reason mod_auth_openidc doesn't check the id_token for distributed claims? Or is there something else I'm missing?

Thanks

**Comment by zandbelt** [comment, marked as answer]
right, there's no good reason for not doing that, just an omission...

**Reply by chris468** [reply]
Thanks for confirming. I started prototyping a fix [here](https://github.com/chris468/mod_auth_openidc/tree/id-token-composite-claims), but probably won't get back to it for a few weeks. It's not working yet - for AAD I think it needs to handle claim sources that don't specify access tokens.
